concurrency-primer.tex

```diff
@@ -706,7 +753,17 @@ \section{Do we always need sequentially consistent operations?}
 they inhibit optimizations that your compiler and hardware would otherwise make.
 
 What if we could avoid some of this slowdown?
-Consider a simple case like the spinlock from \secref{spinlock}.
+Consider a simple case like spinlock.
+\begin{ccode}
```

Comment:
This is not a good example and is not simple. Spinning on RMW operation can cause cache bouncing. Test-and-test-and-set pattern is one way to solve this problem. Functionally, this is correct, but it may mislead readers by oversimplifying spin lock mechanics and ignoring the details behind it.

Spin lock topic should be moved to cache effect section. Here we simply use another example instead. Not sure if scheduling related topic, such as spinning in user space, should be covered though.

Reply:
> Here we simply use another example instead

In 926f645,  spinlock example is replaced with new example provided in section 5.

Reply:
Resolved as a6c68a9 restore the original content back.